src/carbon_sendgrid_adapter.cr

```diff
@@ -126,7 +147,11 @@ class Carbon::SendGridAdapter < Carbon::Adapter
     private def client : HTTP::Client
       @_client ||= HTTP::Client.new(BASE_URI, port: 443, tls: true).tap do |client|
         client.before_request do |request|
-          request.headers["Authorization"] = "Bearer #{api_key}"
+          if authentication_strategy.authenticate_with_basic_auth?
+            client.basic_auth(authentication_strategy.username, authentication_strategy.password)
```

Comment:
https://docs.sendgrid.com/for-developers/sending-email/authentication#basic-authentication